### PR TITLE
Delete solution for users and admin

### LINKED ISSUE #70: Admin feature to delete solution

### PR DESCRIPTION
**Pull request by o-bm**
- User can delete their own solution only
- Admin can delete any solution
close #70 close

### CODE REVIEW

File: client/src/pages/Solution.tsx [hunk 1]
```diff
@@ -27,13 +27,35 @@ const Solution = () => {
...
   const { isAdmin, isLoading } = useCheckRole();
+  const [currentUserId, setCurrentUserId] = useState(null);
 
   useEffect(() => {
     if (id) {
       loadSolution(id, setSolutionData);
       loadComments(id, setComments);
     }
   }, [id]);
+// here
```
Comment (arthurhohinng) [inline client/src/pages/Solution.tsx:38]: No need for this comment?

File: server/controllers/SolutionController.js [hunk 2]
```diff
@@ -128,12 +128,9 @@ exports.edit = async (req, res) => {
...
 };
 
-// TODO: Test this route
+
 exports.delete = async (req, res) => {
   const { id } = req.params;
   await Solution.destroy({ where: { id } });
-  fs.unlink(`${STORAGE_CONFIG.location}/${solution.diagram}`, (err) =>
-    console.log(err),
-  );
```
Comment (arthurhohinng) [inline server/controllers/SolutionController.js:137]: what are these deletions for?

File: client/src/pages/Solution.tsx (hunk 1 above)
Comment (o-bm) [reply client/src/pages/Solution.tsx:38]: yeah I forgot about it

File: server/controllers/SolutionController.js (hunk 2 above)
Comment (o-bm) [reply server/controllers/SolutionController.js:137]: thanks for pointing it out I think I forgot about it when I was changing the delete function.

**Pull request merged**